cmd/argocd/commands/app.go

```diff
@@ -2644,21 +2647,23 @@ func waitOnApplicationStatus(ctx context.Context, acdClient argocdclient.Client,
 			}
 		}
 
+		hydrationFinished := app.Status.SourceHydrator.CurrentOperation != nil && app.Status.SourceHydrator.CurrentOperation.Phase == argoappv1.HydrateOperationPhaseHydrated && app.Status.SourceHydrator.CurrentOperation.SourceHydrator.DeepEquals(app.Status.SourceHydrator.LastSuccessfulOperation.SourceHydrator) && app.Status.SourceHydrator.CurrentOperation.DrySHA == app.Status.SourceHydrator.LastSuccessfulOperation.DrySHA
```

Comment:
I would assume that `app.Status.SourceHydrator` can be null. Missing null check here? A unit test might be missing to catch this case.

Reply:
It can't be nil. I mimicked `SyncStatus`, which is a plain field on `ApplicationStatus` instead of a pointer.  `omitempty` is enabled, so if both fields inside `SourceHydrator` are `nil`, the field won't appear in the k8s API JSON.